### ISSUE TITLE
Finish the apps for Indigo 0.20

### ISSUE THREAD

**Issue by sikthehedgehog** [sikthehedgehog/indigo#7]
Because this issue tracker should reflect how incomplete Indigo is. Will edit this if I notice I missed some task. Also note that some stuff may rely on other features (e.g. I had to implement a drag'n'drop widget).

**New apps:**

- File Cabinet (a file browser)
    - [X] Make a list of strings
    - [X] Read file list from `/` directory
    - [x] Read file list from `/rom` directory
    - [x] Read file list from `/rom` subdirectories
    - [x] Show correct icon for each file
    - [X] Be able to select files
    - [x] Be able to open IXEC files
    - [x] Be able to go into subdirectories
    - [x] Be able to do the two above with double click
    - [x] Be able to go to the parent directory
    - [x] Show current directory

- Solitaire (self-explanatory)
    - [X] Moving cards from stock to heap
    - [X] Moving cards from heap to a pile
    - [X] Moving cards from one pile to another
    - [X] Moving cards from heap to a foundaton
    - [X] Moving cards from a pile to a foundation
    - [ ] Be able to do the two above with double click
    - [ ] Moving cards from a foundation to a pile
    - [X] Winning the game ← the winning animation, that is
    - [X] Restarting the game

**Existing apps:**

- Calculator
    - [X] Keyboard support

- Terminal
    - [X] Keyboard support
    - [X] `cls` (clear screen)
    - [ ] `echo` (write to screen)
    - [X] `pause` (wait for a key)
    - [ ] `set` (set variables) ← may decide to leave it for later
    - [X] `beep` (beeps)

**Comment by middlingsnow77** [comment]
Yeah but you might need support for SC3000 keyboard and a virtual keyboard.

**Comment by sikthehedgehog** [comment]
[SC-3000 keyboard uses a dedicated port](https://www.smspower.org/Development/Keyboard) that doesn't exist on the later systems (it's actually easier to get the Saturn keyboard going because that only requires a passive adapter, or the XBAND keyboard but that one is even harder to get hold of).
